deps/chakrashim/lib/chakra_debug.js

```diff
@@ -0,0 +1,1289 @@
+        function(propertyCategory) {
+          childProperties[propertyCategory].map(function(property) {
+            propertiesArray.push({
+              'name': property['name'],
+              'propertyType': 0,
+              'ref': property['handle']
+            });
+          });
+        });
+      obj['properties'] = propertiesArray;
+    }
+  }
+
+  var DebugManager = (function() {
+    var _internalHandle = Number.MAX_SAFE_INTEGER;
+
+    function GetNextInternalHandle() {
+      return --_internalHandle;
+    }
+
+    var _nextResponseSeq = 0;
+
+    function Response(request) {
+      this.seq = _nextResponseSeq++;
+      this.type = 'response';
+      this.request_seq = request.seq;
+      this.command = request.command;
+      this.success = false;
+      this.running = globalExecutionState ? false : true;
+    }
+
+    var _nextEventSeq = 0;
+
+    function Event(eventName) {
+      this.seq = _nextEventSeq++;
+      this.type = 'event';
+      this.event = eventName;
+      this.success = true;
+      this.running = globalExecutionState ? false : true;
+    }
+
+    var Utility = {
+      MakeResponse: function(request) {
+        return new Response(request);
+      },
+      MakeEvent: function(eventName) {
+        return new Event(eventName);
+      },
+      CreateScopeAndRef: function(scopeType, propertiesArray, index,
+                                    frameIndex) {
+        var scopeRef = GetNextInternalHandle();
+        var scopeAndRef = {};
+        scopeAndRef.scope = {
+          'type': scopeType,
+          'index': index,
+          'frameIndex': frameIndex,
+          'object': {
+            'ref': scopeRef
+          }
+        };
+        var properties = [];
+        propertiesArray.map(function(property) {
+          properties.push({
+            'name': property['name'],
+            'propertyType': 0,
+            'ref': property['handle']
+          });
+        });
+        scopeAndRef.ref = {
+          'handle': scopeRef,
+          'type': 'object',
+          'className': 'Object',
+          'constructorFunction': {
+            'ref': globalExecutionState.GetObjectEvalHandle()
+          },
+          'protoObject': {
+            'ref': globalExecutionState.GetObjectProtoEvalHandle()
+          },
+          'prototypeObject': {
+            'ref': globalExecutionState.GetObjectPrototypeEvalHandle()
+          },
+          'properties': properties
+        };
+
+        return scopeAndRef;
+      }
+    };
+
+    var _breakpoints = [];
+    var _pendingBreakpoints = [];
+
+    var BreakpointManager = {
+      GetBreakpoints: function() {
+        var breakpoints = CallHostFunction(chakraDebug.JsDiagGetBreakpoints);
+        var activeBreakpointList = [];
+        breakpoints.forEach(function(element, index, array) {
+          var bpId = element.breakpointId;
+          activeBreakpointList[bpId] = _breakpoints[bpId];
+        });
+        _breakpoints = activeBreakpointList;
```

Comment:
Nit: can be a little more succinct doing _breakpoints = breakpoints.map(function(element, index, array) { return element.breakpointId; })